CHANGELOG.md

```diff
@@ -4,6 +4,9 @@
 - Shuffle `IterableExtension.sample` results.
 - Fix `mergeSort` when the runtime iterable generic is a subtype of the static
   generic.
+- Remove `firstOrNull`, `lastOrNull`, `singleOrNull` and `elementAtOrNull()`
```

Comment:
If these are deprecated, it is impossible to switch to the non-deprecated version. You can't switch from `.firstOrNull` to `.firstOrNull`.
A removal is fully seamless, on the other hand. Could you check the internal discussion that I CC'd you on

Reply:
Oh! I guess as long as the SDK constraint is correct.

I *think* we'll still need a breaking change version, though. @natebosch ? @devoncarew ?

Reply:
> A removal is fully seamless, on the other hand

This is probably true for the vast majority of code. Code with `import 'dart:core' as core;` can have trouble here, but that's pretty rare outside of generated code, and most  also have an unprefixed `import 'dart:core';`.

We could consider exporting the `dart:core` extension here. That should keep it non-breaking for even the edge cases. I think the analyzer will still have a diagnostic - but it might switch from unused import to "all used symbols from this import are also available from another import".

WDYT @lrhn

Reply:
I'd *prefer* to not export the extension from `dart:core`. 
We can, but I don't think it'll ever bring us anything but hardache.

And do it in a major version increment release, just to be technically allowed to be breaking, even if we hope that nobody notices.

It's not entirely non-breaking, even if we expose extensions with the same signatures, `on` types and name.
If someone uses `show` or `hide` of `IterableExtension` for a reason, or use explicit extension invocation, then it can break when that extension no longer has the members.
Unlikely, not impossible.

At least we dodge the bullet of having the extension names themeslves conflict, because this one is called `IterableExtension` and the one in `dart:collection` is called <code>IterableExtension**s**</code>. 
So we *could* export it. But let's not take that kind of chances.
(And thank deity for lack of consistency.)